lib/nerves_hub/fwup.ex

```diff
@@ -47,25 +45,25 @@ defmodule NervesHub.Fwup do
     end
   end
 
-  defp metadata_value(metadata, key) when is_binary(key) do
-    {:ok, regex} = "#{key}=\"(?<value>[^\n]+)\"" |> Regex.compile()
+  defp parse_metadata(metadata) do
+    Regex.scan(@metadata_regex, metadata)
```

Comment:
```suggestion
    Regex.scan(~r/meta-(?<key>[^\n]+)=\"(?<value>[^\n]+)\"/, metadata)
```
`~r` is precompiled so the regex can be here if you like that better.